meson.build

```
@@ -279,10 +281,11 @@ endif
 conf.set10('HAVE_STATIC_LIBCAP', libcap_static.found())
 
 ## PAM.
-pam = cc.find_library('pam',
-    has_headers: 'security/pam_modules.h')
-conf.set10('HAVE_PAM', pam.found())
-pkgconfig_libs += pam
+if want_pam_cgroup == true
+	pam = cc.find_library('pam', has_headers: 'security/pam_modules.h', required: true)
+	conf.set10('HAVE_PAM', pam.found())
+	pkgconfig_libs += pam
+endif
```

The `pam-cgroup` option is a boolean, and this block is only entered if it is force enabled. In this case, you make finding the library mandatory (the `required: true` kwarg is the implicit default), and then you set the HAVE_PAM define based on the value of `pam.found()`... which is impossible to ever be not found, because it was required on the line above and setup will simply error out if it is not found.

If you want to actually make this optional, you may want to switch the option to a feature option, and use:


```suggestion
pam = cc.find_library('pam', has_headers: 'security/pam_modules.h', required: get_option('pam-cgroup'))
conf.set10('HAVE_PAM', pam.found())
pkgconfig_libs += pam
```

---

Thank you!